subsys/logging/log_core.c

```diff
@@ -50,7 +50,7 @@ LOG_MODULE_REGISTER(log);
 struct log_strdup_buf {
 	atomic_t refcount;
 	char buf[CONFIG_LOG_STRDUP_MAX_STRING + 1]; /* for termination */
-};
+} __aligned(sizeof(uintptr_t));
```

Comment:
don't you need to add padding after refcount which is `int` to ensure that buf is `uintptr_t` aligned?

Reply:
In `kernel/mem_slab.c` the function `create_free_list()` has this check
```
	CHECKIF(((slab->block_size | (uintptr_t)slab->buffer) &
				(sizeof(void *) - 1)) != 0U) {
```
where the block_size is the `sizeof(log_strdup_buf)` and the buffer is the start of the `log_strdup_pool_buf`. Without this alignment, the result of this CHECKIF(4 != 0U) and then the init fails. After I added the uintptr_t alignment, this check passes and the `k_mem_slab_init()` succeeds.

Reply:
i think that it should be :
```
struct log_strdup_buf {
	atomic_t refcount;
	char buf[CONFIG_LOG_STRDUP_MAX_STRING + 1] __aligned(sizeof(uintptr));
};
```

I wrote a simple test for `qemu_x86_64` and in fact `k_mem_slab_init` was failing when struct was aligned and passing when `buf` was aligned.